python/ray/data/dataset.py

```diff
@@ -2638,7 +2653,36 @@ def gen():
 
         class Iterable:
             def __init__(self, blocks, epoch):
-                self._splits = blocks.split(split_size=blocks_per_window)
+                if bytes_per_window:
+                    self._splits = blocks.split_by_bytes(bytes_per_window)
+                else:
+                    self._splits = blocks.split(split_size=blocks_per_window)
+                try:
+                    sizes = [s.size_bytes() for s in self._splits]
```

Comment:
size_bytes() returns -1 if the size is unknown instead of exception? maybe just `assert min(sizes) >= 0`?